CycleSort/Cycle_Sort.cpp

```diff
@@ -0,0 +1,94 @@
+/* 
+Cycle Sort
+Cycle sort is a comparison based sorting algorithm which forces array to be factored into 
+the number of cycles where each of them can be rotated to produce a sorted array.
+It is an in-place and unstable sorting algorithm. 
+Best and Average case time complexity:O(n^2)
+
+
+Algorithm
+An element a is given, index of a can be calculated by counting the number of elements that 
+are smaller than a.
+1. If the element is found to be at its correct position, simply leave it as it is.
+2. Otherwise, find the correct position of a by counting the total number of elements that are 
+less than a. where it must be present in the sorted array. The other element b which is replaced is 
+to be moved to its correct position. This process continues until we got an element at the 
+original position of a. */
```

Comment:
@dldisha instead of adding and commenting out a long description of this algorithm, I'll suggest you to make a README.md file, with proper explanation, algorithm, pseudocode, time complexity etc. It can fetch you more points.
Your code looks good though.